src/expr/src/scalar/func.rs

```diff
@@ -1009,6 +1009,19 @@ fn power_numeric<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError>
     }
 }
 
+fn get_byte<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let bytes = a.unwrap_bytes();
+    let index = b.unwrap_int32();
+    let err = EvalError::IndexOutOfRange {
+        provided: index,
+        valid_end: i32::try_from(bytes.len()).unwrap() - 1,
```

Comment:
For zero length `bytes` this may go negative and be confusing. Would it make sense to have `len` instead, and report in the error that the index must be strictly less than this?

Reply:
Nevermind! Postgres reports a valid length of -1, just as you have done / tested.